test/single_async_manual_reset_event_test.cpp

```diff
@@ -0,0 +1,421 @@
+using unifex::just;
+using unifex::schedule;
+using unifex::single_thread_context;
+using unifex::start;
+using unifex::sync_wait;
+using unifex::tag_t;
+using unifex::transform;
+using unifex::transform_done;
+using unifex::with_query_value;
+
+namespace {
+
+struct mock_receiver_impl {
+  MOCK_METHOD(void, set_value, (), ());
+  MOCK_METHOD(void, set_error, (std::exception_ptr), (noexcept));
+  MOCK_METHOD(void, set_done, (), (noexcept));
+};
+
+// mock_receiver_impl cannot be used directly as a receiver because the MOCK
+// macros make the type non-movable, non-copyable. Receivers must be movable.
+struct mock_receiver {
+  mock_receiver(inline_scheduler& scheduler, inplace_stop_source& stopSource)
+      : impl(std::make_unique<mock_receiver_impl>()),
+        scheduler(&scheduler),
+        stopToken(stopSource.get_token()) {}
+
+  void set_value() {
+    impl->set_value();
+  }
+
+  void set_error(std::exception_ptr e) noexcept {
+    impl->set_error(e);
+  }
+
+  void set_done() noexcept {
+    impl->set_done();
+  }
+
+  std::unique_ptr<mock_receiver_impl> impl;
+  inline_scheduler* scheduler;
+  inplace_stop_token stopToken;
+
+  friend inline_scheduler tag_invoke(
+      tag_t<get_scheduler>, const mock_receiver& self) noexcept {
+    return *self.scheduler;
+  }
+
+  friend inplace_stop_token tag_invoke(
+      tag_t<get_stop_token>, const mock_receiver& self) noexcept {
+    return self.stopToken;
+  }
+};
+
+} // namespace
+
+struct single_async_manual_reset_event_test : testing::Test {
+  inplace_stop_source stopSource;
+  inline_scheduler scheduler;
+  mock_receiver receiver{scheduler, stopSource};
+  mock_receiver_impl& receiverImpl = *receiver.impl;
+};
+
+TEST_F(single_async_manual_reset_event_test, default_constructor_leaves_primitive_unready) {
+  single_async_manual_reset_event evt;
+
+  EXPECT_FALSE(evt.ready());
+}
+
+TEST_F(single_async_manual_reset_event_test, can_construct_initially_ready_primitive) {
+  single_async_manual_reset_event evt{true};
+
+  EXPECT_TRUE(evt.ready());
+}
+
+TEST_F(single_async_manual_reset_event_test, set_makes_unready_primitive_ready) {
+  single_async_manual_reset_event evt;
+
+  evt.set();
+
+  EXPECT_TRUE(evt.ready());
+}
+
+TEST_F(single_async_manual_reset_event_test, reset_makes_ready_primitive_unready) {
+  single_async_manual_reset_event evt{true};
+
+  evt.reset();
+
+  EXPECT_FALSE(evt.ready());
+}
+
+TEST_F(single_async_manual_reset_event_test, sender_completes_after_set_when_connected_to_unready_primitive) {
+  single_async_manual_reset_event evt;
+
+  auto op = connect(evt.async_wait(), std::move(receiver));
+
+  {
+    EXPECT_CALL(receiverImpl, set_value()).Times(0);
+    EXPECT_CALL(receiverImpl, set_error(_)).Times(0);
+    EXPECT_CALL(receiverImpl, set_done()).Times(0);
+
+    start(op);
+  }
+
+  EXPECT_CALL(receiverImpl, set_value()).Times(1);
+  EXPECT_CALL(receiverImpl, set_error(_)).Times(0);
+  EXPECT_CALL(receiverImpl, set_done()).Times(0);
```

Comment:
I forget - is the `EXPECT_CALL(...).Times(1)` checking that "There have so far been 1 call to this method" or "There will eventually be exactly 1 call to this method by the time the mock object destructs"?
The usage seems like the former, but my recollection of mocking frameworks was that many of the methods behave like the latter.